dfsclient/src/main/java/org/hpccsystems/dfs/client/ColumnPruner.java

```diff
@@ -141,6 +141,7 @@ public FieldDef pruneRecordDefinition(FieldDef originalRD) throws Exception
 
         FieldDef ret = new FieldDef(originalRD);
         ret.setDefs(selectedFields.toArray(new FieldDef[0]));
+        ret.setDataLen(0);
```

Comment:
this is scary, let's add a comment regarding the net effect of this line @jpmcmu